metadata-db-app/src/cmr/metadata_db/data/oracle/concepts.clj

```diff
@@ -224,6 +224,80 @@
        :existing-concept-id concept_id
        :existing-native-id native_id})))
 
+(defn validate-collection-not-associated-with-another-variable-with-same-name
+  "Validates that collection in the concept is not associated with a different
+  variable, which has the same name as the variable in the concept.
+  Returns nil if valid and an error response if invalid."
+  [db concept]
+  (let [variable-concept-id (get-in concept [:extra-fields :variable-concept-id])
+        associated-concept-id (get-in concept [:extra-fields :associated-concept-id])]
+    (when (and variable-concept-id associated-concept-id))
+      ;; Get all the other variables sharing the same name; not deleted
+      ;; and are associated with the same collection.
+      (let [stmt [(format "select va.variable_concept_id
+                           from   cmr_variable_associations va, 
+                                  (select concept_id, max(revision_id) maxrev
+                                   from cmr_variable_associations group by concept_id) latestva,
+                                  cmr_variables v
+                           where  va.revision_id = latestva.maxrev
```

Comment:
I personally find joins more readable when determining the relationship between tables, you can keep it this way if you like but I do think its more maintainable when you delineate the where from the join.

Reply:
I'm much more comfortable with the from/where.  Also, not sure how the subquery table will work with joins.  Oracle syntax seems much more strict than sybase, and not that intuitive, to me.